core/src/main/scala/org/apache/spark/deploy/security/HiveCredentialProvider.scala

```diff
@@ -29,11 +29,10 @@ import org.apache.hadoop.io.Text
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
 import org.apache.hadoop.security.token.Token
 
-import org.apache.spark.SparkConf
 import org.apache.spark.internal.Logging
 import org.apache.spark.util.Utils
 
-private[security] class HiveCredentialProvider extends ServiceCredentialProvider with Logging {
+private[security] class HiveCredentialProvider extends HadoopDelegationTokenProvider with Logging {
```

Comment:
Right, those deps should be in test scope, but I ran into trouble doing that: https://github.com/apache/spark/pull/17723/files#r112788867

I'll go back and try again.  Suggestions welcome.